### PR TITLE
Change key to trigger rebuild completion word list

### LINKED ISSUE #1107: Word Completion plugin shortcut conflicts with the shortcut for toggling the sidebar
### What Happened?

The [Word Completion plugin](https://github.com/elementary/code/tree/master/plugins/word-completion) uses `Ctrl` `\` for [rebuilding the word list](https://github.com/elementary/code/wiki/shortcuts#plugins---word-completion). However, this shortcut now toggles the sidebar. 

### Steps to Reproduce

1. Enable the Word Completion plugin.
2. Type a word that does not appear in the current document.
2. Enter `Ctrl` `\`.
3. The sidebar will open/close. The word list is not rebuilt.

### Expected Behavior

There should be a different shortcut that doesn't conflict with existing ones. I _think_ `Ctrl` `/` would be fine and just deleting "back" in the line below should work, but I don't have the wherewithal to test it myself.

https://github.com/elementary/code/blob/41b068e5776e78bfa096744ca460f988e002b841/plugins/word-completion/plugin.vala#L41

### OS Version

6.x (Odin)

### Software Version

Latest release (I have run all updates)

### Log Output

_No response_

### Hardware Info

_No response_

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #1107 

The shortcut to trigger a rebuild of the completion word list is changed to `<Ctrl>|`.  On English keyboards at least this is close to the original `<Ctrl>\`, requiring only to hold down the <Shift> key in addition.

Not sure how to expose this shortcut, but that can wait for another PR.

### CODE REVIEW

File: plugins/word-completion/plugin.vala [hunk 1]
```diff
@@ -39,7 +39,7 @@ public class Scratch.Plugins.Completion : Peas.ExtensionBase, Peas.Activatable {
         Gdk.Key.ISO_Left_Tab,
     };
 
-    private const uint USER_REQUESTED_KEY = Gdk.Key.backslash;
+    private const string REFRESH_SHORTCUT = "|"; //In combination with <Ctrl> will cause refresh
```
Comment (danrabbit) [inline plugins/word-completion/plugin.vala:42]: Is this Gdk.Key.vertbar ?

Comment (jeremypw) [reply plugins/word-completion/plugin.vala:42]: No, it is Gdk.Key.bar  (strangely).  I'll use that instead.

**Comment by jeremypw** [comment]
Not sure how to expose the changed shortcut, putting it in the provider header is a possibility but since we cannot style the header (afaik) the appearance is not good:

Screenshot from 2022-09-02 09 25 46

**Review by danrabbit — APPROVED** [review]
No idea how to expose this shortcut either, or how useful it is. Probably can leave that to another PR. This closes the linked issue :)

**Pull request merged**